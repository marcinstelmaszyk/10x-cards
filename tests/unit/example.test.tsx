import { describe, it, expect, vi } from "vitest";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

// This is just an example test file to demonstrate structure
// Replace this with your actual component tests

describe("Example component test", () => {
  it("renders correctly", () => {
    // Example of a simple render test
    render(<div data-testid="example">Hello World</div>);
    expect(screen.getByTestId("example")).toBeInTheDocument();
    expect(screen.getByText("Hello World")).toBeVisible();
  });

  it("handles user interactions", async () => {
    // Example of testing user interactions
    const user = userEvent.setup();
    const mockFn = vi.fn();

    render(
      <button data-testid="test-button" onClick={mockFn}>
        Click me
      </button>
    );

    await user.click(screen.getByTestId("test-button"));
    expect(mockFn).toHaveBeenCalledOnce();
  });

  it("matches snapshot", () => {
    // Example of a snapshot test
    const { container } = render(<div>Snapshot test example</div>);
    expect(container).toMatchSnapshot();
  });
});
