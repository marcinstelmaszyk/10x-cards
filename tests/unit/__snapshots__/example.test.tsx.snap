// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Example component test > matches snapshot 1`] = `
<div>
  <div>
    Snapshot test example
  </div>
</div>
`;
